wagtail/wagtailadmin/views/account.py

```diff
@@ -141,3 +141,16 @@ def logout(request):
     request.session.modified = False
 
     return response
+
+
+def change_avatar(request):
+    if request.method == 'POST':
+        form = AvatarPreferencesForm(request.POST, request.FILES, instance=UserProfile.get_for_user(request.user))
+        if form.is_valid():
+            form.save()
+            messages.success(request, _("Your preferences have been updated successfully!"))
+            return redirect('wagtailadmin_account_change_avatar')
+    else:
+        form = AvatarPreferencesForm(instance=UserProfile.get_for_user(request.user))
+
+    return render(request, 'wagtailadmin/account/change_avatar.html', {'form': form})
```

Comment:
All the logic part from this view is already written in the generic `FormView`, except the success message and the `get_for_user` part.
It would be better if you used it instead: you can put the success message to `form_valid(…)` and `get_for_user` in `get_object(…)`.